--- monero_headers.h	2020-07-04 09:45:31.170959000 +0200
+++ monero_headers.h.xhv	2020-07-04 09:49:23.878959400 +0200
@@ -9,9 +9,9 @@
 #define BLOCKCHAIN_DB DB_LMDB
 
 
-#define UNSIGNED_TX_PREFIX "Monero unsigned tx set\003"
-#define SIGNED_TX_PREFIX "Monero signed tx set\003"
-#define KEY_IMAGE_EXPORT_FILE_MAGIC "Monero key image export\002"
+#define UNSIGNED_TX_PREFIX "Haven unsigned tx set\003"
+#define SIGNED_TX_PREFIX "Haven signed tx set\003"
+#define KEY_IMAGE_EXPORT_FILE_MAGIC "Haven key image export\002"
 #define OUTPUT_EXPORT_FILE_MAGIC "Monero output export\003"
 
 #define FEE_ESTIMATE_GRACE_BLOCKS 10
